src/registrar/assets/js/get-gov.js

```diff
@@ -232,14 +246,88 @@ function handleValidationClick(e) {
   for(const input of needsValidation) {
     input.addEventListener('input', handleInputValidation);
   }
+  const alternativeDomainsAvailability = document.getElementById('check-avail-for-alt-domains');
   const activatesValidation = document.querySelectorAll('[validate-for]');
+
   for(const button of activatesValidation) {
-    button.addEventListener('click', handleValidationClick);
+    // Adds multi-field validation for alternative domains
+    if (button === alternativeDomainsAvailability) {
+      button.addEventListener('click', (e) => {
+        handleFormsetValidationClick(e, alternativeDomainsAvailability)
+      });
+    } else {
+      button.addEventListener('click', handleValidationClick);
+    }
   }
+
+  
+  // Add event listener to the "Check availability" button
+  const checkAvailabilityButton = document.getElementById('check-availability-button');
+  if (checkAvailabilityButton) {
+    const targetId = checkAvailabilityButton.getAttribute('validate-for');
+    const checkAvailabilityInput = document.getElementById(targetId);
+    checkAvailabilityButton.addEventListener('click',
+      function() { 
+        removeFormErrors(checkAvailabilityInput, true);
+      }
+    );
+  }
+
+  // Add event listener to the alternate domains input
+  const alternateDomainsInputs = document.querySelectorAll('[auto-validate]');
```

Comment:
This block should be obsolete

Reply:
I thought so too bc we're using `validate-for`, but I think somethings enmeshed with `auto-validate` bc when I remove this block all the buttons (check availability, add alternative domain), etc break

Reply:
AH turns out we were duplicating the work of the two onclick buttons + availability button hence why the removal. Great catch Rachid + thank you Erin!!